### PR TITLE
fix(tooltip): floating tooltip

### PR DESCRIPTION
**Pull request by dreamwasp**
### Overview

Floating ToolTips were failing axe tests because they had a 'labelled-by' trait that was referencing the Popover component when the Popover was closed + out of the DOM.

### PR Checklist

- [ ] Related to designs:
- [x] Related to JIRA ticket: WEB-2046
- [x] I have run this code to verify it works
- [x] This PR includes unit tests for the code change
- [x] This PR includes testing instructions tests for the code change
- [x] The alpha package of this PR is passing end-to-end tests in all relevant Codecademy repositories

#### Testing Instructions

> Test that PopoverTooltips still sound as expected

#### PR Links and Envs

| Repository   | PR Link                                                  | PR Env                                                   |
| :----------- | :------------------------------------------------------- | :------------------------------------------------------- |
| Monolith     | [Monolith PR](https://github.com/codecademy-engineering/Codecademy/pull/31882)  | [Monolith Env](https://pr-31882-monolith.dev-eks.codecademy.com/) |
| Portal       | [Portal Link](https://github.com/codecademy-engineering/portal-app/pull/2324)  | [Portal Env](https://tengu.codecademy.com/catalog?PR_ENV=portal-app-pr-2324)   |

### CODE REVIEW

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.169-alpha.382075.0<br/>@codecademy/gamut-labs@33.0.5-alpha.382075.0<br/>@codecademy/gamut@49.1.1-alpha.382075.0<br/>@codecademy/monaco@0.6.64-alpha.382075.0<br/>@codecademy/styleguide@60.1.1-alpha.382075.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://62e8098c09307f008a998470--gamut-preview.netlify.app>https://62e8098c09307f008a998470--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/62e8098c09307f008a998470>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.169-alpha.8c03fb.0<br/>@codecademy/gamut-labs@33.0.5-alpha.8c03fb.0<br/>@codecademy/gamut@49.1.1-alpha.8c03fb.0<br/>@codecademy/monaco@0.6.64-alpha.8c03fb.0<br/>@codecademy/styleguide@60.1.1-alpha.8c03fb.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://62e81f9a16e11e0f7b0f820d--gamut-preview.netlify.app>https://62e81f9a16e11e0f7b0f820d--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/62e81f9a16e11e0f7b0f820d>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.169-alpha.d95b39.0<br/>@codecademy/gamut-labs@33.0.5-alpha.d95b39.0<br/>@codecademy/gamut@49.1.1-alpha.d95b39.0<br/>@codecademy/monaco@0.6.64-alpha.d95b39.0<br/>@codecademy/styleguide@60.1.1-alpha.d95b39.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://62e8242638e8de13cca7c059--gamut-preview.netlify.app>https://62e8242638e8de13cca7c059--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/62e8242638e8de13cca7c059>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-icons@7.5.0-alpha.902fa0.0<br/>@codecademy/gamut-kit@0.6.169-alpha.902fa0.0<br/>@codecademy/gamut-labs@33.0.5-alpha.902fa0.0<br/>@codecademy/gamut@49.1.1-alpha.902fa0.0<br/>@codecademy/monaco@0.6.64-alpha.902fa0.0<br/>@codecademy/styleguide@60.1.1-alpha.902fa0.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://62e82bcf038c081a0e0e50ab--gamut-preview.netlify.app>https://62e82bcf038c081a0e0e50ab--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/62e82bcf038c081a0e0e50ab>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-icons@7.5.0-alpha.f08b67.0<br/>@codecademy/gamut-kit@0.6.169-alpha.f08b67.0<br/>@codecademy/gamut-labs@33.0.5-alpha.f08b67.0<br/>@codecademy/gamut@49.1.1-alpha.f08b67.0<br/>@codecademy/monaco@0.6.64-alpha.f08b67.0<br/>@codecademy/styleguide@60.1.1-alpha.f08b67.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://62e934d6c6bc750f02716655--gamut-preview.netlify.app>https://62e934d6c6bc750f02716655--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/62e934d6c6bc750f02716655>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.171-alpha.ad8e1e.0<br/>@codecademy/gamut-labs@33.0.7-alpha.ad8e1e.0<br/>@codecademy/gamut@50.0.1-alpha.ad8e1e.0<br/>@codecademy/monaco@0.6.66-alpha.ad8e1e.0<br/>@codecademy/styleguide@61.0.1-alpha.ad8e1e.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://62e976431e5c9d4708e5a036--gamut-preview.netlify.app>https://62e976431e5c9d4708e5a036--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/62e976431e5c9d4708e5a036>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.172-alpha.f50784.0<br/>@codecademy/gamut-labs@33.0.8-alpha.f50784.0<br/>@codecademy/gamut@50.1.1-alpha.f50784.0<br/>@codecademy/monaco@0.6.67-alpha.f50784.0<br/>@codecademy/styleguide@61.1.1-alpha.f50784.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://62ec19d34d1e7c1e2cc704b6--gamut-preview.netlify.app>https://62ec19d34d1e7c1e2cc704b6--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/62ec19d34d1e7c1e2cc704b6>Deploy Logs</a></p>

File: packages/gamut/src/ToolTip/utils.tsx [hunk 1]
```diff
@@ -79,14 +79,33 @@ export const escapeKeyPressHandler = (
...
+  const ariaLabel = {
+    'aria-labelledby':
+      isOpenPopoverToolTip === undefined || isOpenPopoverToolTip
+        ? id
+        : undefined,
+    'aria-label':
+      isOpenPopoverToolTip === undefined || isOpenPopoverToolTip
+        ? undefined
+        : id,
+  };
```
Comment (BandanaKM) [inline packages/gamut/src/ToolTip/utils.tsx:104]: In order to make this more succinct, what if we did:

```
const ariaLabel = isOpenPopoverToolTip === undefined || isOpenPopoverToolTip ? { 'aria-labelledby': id } : { 'aria-label': id }
```
And kept the same destructing below

Comment (aresnik11) [reply packages/gamut/src/ToolTip/utils.tsx:104]: i like this! or if we need to set the opposite case as undefined then we can pull out `isOpenPopoverToolTip === undefined || isOpenPopoverToolTip` into a shared variable

**Review by BandanaKM — APPROVED** [review]
Looks good! Thanks @judahtanthony

**Review by judahtanthony — APPROVED** [review]
This fixed it in my branch https://github.com/codecademy-engineering/portal-app/pull/2320

**Pull request merged**